### ISSUE TITLE
``rebin`` does not modify the rebinned axes offsets

### ISSUE THREAD

**Issue by francisco-dlp** [hyperspy/hyperspy#1846]
For example, currently:

```python

>>> import hyperspy.api as hs
>>> import numpy as np
>>> s = hs.signals.Signal1D(np.arange((10)))
>>> sb = s.rebin((5,))
>>> sb.metadata.Signal.binned
False
>>> sb.axes_manager.signal_axes[0].offset
0.0
>>> s.axes_manager.signal_axes[0].offset
0.0
```

The expected behaviour is:

```python

>>> import hyperspy.api as hs
>>> import numpy as np
>>> s = hs.signals.Signal1D(np.arange((10)))
>>> sb = s.rebin((5,))
TypeError: The signal is not binned, hence it cannot be rebinned
>>> sb.metadata.Signal.binned = True
>>> sb = s.rebin((5,))
>>> sb.axes_manager.signal_axes[0].offset
0.5
>>> s.axes_manager.signal_axes[0].offset
0.0
```

**Comment by thomasaarholt** [comment]
Nice catch with not being able to rebin a non-binned signal.

**Comment by k8macarthur** [comment]
So if I get this error correctly the offset should be changed so that each value sits in the middle of the bin?

**Comment by k8macarthur** [comment]
I'm trying to work on the rebin function at the moment but I don't quite understand the problem here...
Could anyone explain more clearly about the offset? Also I'm guessing the 'binned' search should go in                                         each of the signal classes and not in the base function right?

**Comment by francisco-dlp** [comment]
Imagine that you are binning a 3 channels spectrum image into just one channel. If the first pixel energy is centred at 0 eV, second at 1 eV and third at 2 eV, the result of binning should be a single 3x larger pixel centred at 1 eV. However, currently we get a 3x larger pixel centred at 0 eV. The counts and scale are correctly computed, only the offset is wrong.

I think that both the "binned" and "offset" bugs should be fixed in the parent method.

**Comment by k8macarthur** [comment]
Ok I've updated #1958 with these two corrections as well.

**Comment by k8macarthur** [comment]
Assuming the new offset is (n-1)*x/2 where n is the rebin scale and x is the original scale.

**Comment by ericpre** [comment]
Fixed in #1965.
